llms/mlx_lm/tuner/lora.py

```diff
@@ -85,7 +85,7 @@ def __init__(
         self.dropout = nn.Dropout(p=dropout)
 
         # Scale for low-rank update
-        self.scale = scale * (alpha / r)
+        self.scale = mx.array(scale * (alpha / r))
```

Comment:
maybe we need to add ` mx.stop_gradient(self.scale)`

Reply:
```
self.freeze(keys=["scale"], recurse=False)
```

Reply:
Also I find it a bit odd that we have an `alpha` and a `scale` parameter since they are both just multiplicative scales 🤔 . I guess that was the original LoRA parameterization but maybe it is worth simplifying it to a single scale parameter.

Reply:
Yeah, the original LoRa paper does not have the scale parameter (https://github.com/microsoft/LoRA/blob/main/loralib/layers.py#L112), but either way, we need to store the final scale parameter in the weight so that it can be retrieved from the adapter. Alternatively, we could save the LoRa configuration along with the weight.

Reply:
I think the way you've done it is fine in terms of making it an `mx.array`.

I would like to merge `alpha` and `scale` into a single parameter to simplify the number of inputs. But that can be done in a follow up.

Reply:
I was actually looking at that earlier in comparison to the LoRA paper and when considering which of the two to include in hyperparameter sweeps.  Merging them would definitely simplify things

Reply:
>  To address this issue, we added the scale parameter. I am not sure why using only r and alpha caused issues in fusing models,

We needed a way to increase the size of the adapters weights so they didn't get erased in fusion, which is why I added the [`scale`](https://github.com/ml-explore/mlx-examples/pull/252/files/8b84afa63900cf3697776e97f20d1b98b055398b#r1445442551). At the time there was no `alpha` parameter, we just fixed it at `2` (which is what the LoRA paper does IIRC). In hindsight I should have just named `scale` `alpha`. Somewhere along the way we added added an additional `alpha` parameter. I don't think it makes sense to include both, the `scale` is really the same as the `alpha`.

Reply:
Its quite reasonable to have quite different `alpha` parameters. Keeping only `alpha` and `r` will be more consistent with what other frameworks do IMO. The `scale` is the oddball here.

Reply:
Btw looking at [that thread](https://github.com/ml-explore/mlx-examples/pull/252/files/8b84afa63900cf3697776e97f20d1b98b055398b#r1445442551), maybe we should experiment with other initializations. Even with a large scale, we still see issues with fusion from time to time.

Reply:
+1 for removing scale. alpha + r is better.

Reply:
The only concern I have is that using alpha normally creates a scale value around 2 or 3, but in the current alpha+ scale we are getting around 20 to 30 scaling. I'm not sure if that will make a significant difference in output or training.

Reply:
I actually ran into this yesterday I have a LoRA adapter trained with a scale of 1, but when running a benchark, the results were worse than the base model. When I found the problem, I forgot to set the scale back to 1 from the default when running the model for the benchmark.